docs/references/unittest.rst

```diff
@@ -0,0 +1,179 @@
+Unit Test Framework
+========================
+
+.. note::
+	Available in v0.11.0 and later
+
+Overview
+*********
+
+You can write test cases for your tool by defining the ``tests()`` function in your Tool class.
+Test cases defined by this function will be picked up by ``janisdk run-test`` command.
+
+This unit test framework provides several predefined preprocessors to
+transform Janis execution output data in the format that can be tested.
+For example, there is a preprocessor to read the md5 checksum of an output file.
+In addition to the predefined preprocessors, the frameworks also allows users to define and pass their own preprocessors.
+
+
+Define test cases
+-----------------
+You can define multiple test cases per tool.
+For each test case, you can declare multiple expected outputs.
+Mostly, you really only need to define more test cases if they require different input data.
+
+
+.. code-block:: python
+
+    class BwaAligner(BioinformaticsWorkflow):
+        def id(self):
+            return "BwaAligner"
+
+        ...
+
+        def tests(self):
+            return [
+                TTestCase(
+                    name="basic",
+                    input={
+                        "bam": "https://some-public-container/directory/small.bam"
```

Comment:
Would be worth stating that for tests, remote files (like those that specify a HTTP link) will get localised to the config directory. Maybe just something like:

> `janisdk` will automatically localise the remote file for execution

Reply:
Very good point. Thanks! I have added a section that talks about test files https://github.com/PMCC-BioinformaticsCore/janis/pull/35/files#diff-4e058bfce2b1010ca6627509a5f1c776e1795caf2ec5a0ffd33f4c7bd5cc325eR85-R107